src/intel_rdt.c

```diff
@@ -164,6 +166,18 @@ static void rdt_submit(const struct pqos_mon_data *group) {
   }
 #endif
 
+  if (events & PQOS_PERF_EVENT_LLC_MISS) {
+#if PQOS_VERSION >= 40400
+    uint64_t value;
+
+    int ret = pqos_mon_get_value(group, PQOS_PERF_EVENT_LLC_MISS, &value, NULL);
+    if (ret == PQOS_RETVAL_OK)
+      rdt_submit_gauge(desc, "bytes", "llc_miss", value);
+#else
```

Comment:
Looking at intel-cmt-cat Git history...

Whereas  `pqos_mon_get_value()` was added 2 years ago for in 4.04, same time with LLC refs support:
https://github.com/intel/intel-cmt-cat/commit/6ee8ee7fddce835e3e387a40b027c7d0f34d4462

LLC misses counter support was added alread >8 years ago, for version 1.03:
https://github.com/intel/intel-cmt-cat/commit/53261fa6ce2be876a424e27c95b7e8941b40b98b

As the other code in this function just takes numbers directly from `values` (i.e. bypasses locking used by  `pqos_mon_get_value()`), and delta parameter is not used, what value there's from having this ifdef for calling  `pqos_mon_get_value()`?

Reply:
No update will be required if pqos_mon_data structure will change

Reply:
Update is still needed for the `#else` branch in that case...

Reply:
I was suggesting using just the code in `#else` branch, similarly to other code in here, unless there's compelling reason to do otherwise.

If lock checks done in them is one, it would be better to have **all** the values requested with `pqos_*()` getters inside `#if PQOS_VERSION >= 40400`, and `#else` branch doing the same for all values (that are available before 4.04) by accessing the struct values directly.

Reply:
done